File: app/src/main/java/com/w9jds/marketbot/classes/modules/ApplicationModule.java
package com.w9jds.marketbot.classes.modules;

import android.app.Application;
import android.content.SharedPreferences;
import android.database.sqlite.SQLiteDatabase;
import android.preference.PreferenceManager;

import com.w9jds.marketbot.classes.StorageScope;
import com.w9jds.marketbot.data.Database;

import javax.inject.Named;
import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class ApplicationModule {

    private Application application;

    public ApplicationModule(Application application) {
        this.application = application;
    }

    @Provides
    @Singleton
    Application provideApplication() {
        return application;
    }

    @Provides
    @Singleton
    SharedPreferences provideSharedPreferences(Application application) {
        return PreferenceManager.getDefaultSharedPreferences(application);
    }

}

File: app/src/main/java/com/w9jds/marketbot/classes/modules/ApplicationModule.kt
package com.w9jds.marketbot.classes.modules

import android.app.Application
import android.arch.persistence.room.Room
import android.content.SharedPreferences
import android.preference.PreferenceManager
import com.google.firebase.database.FirebaseDatabase
import com.w9jds.marketbot.data.Database
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class ApplicationModule(private val application: Application) {

//    private val firebase: FirebaseDatabase = FirebaseDatabase.getInstance()
//    private val database: Database = Room.databaseBuilder(application,
//            Database::class.java, "market-storage").build()

    @Provides
    @Singleton
    fun providesApplication(): Application = application

    @Provides
    @Singleton
    fun provideSharedPreferences(): SharedPreferences = PreferenceManager.getDefaultSharedPreferences(application)

    @Provides
    @Singleton
    fun provideFirebaseDatabase(): FirebaseDatabase {
        val database: FirebaseDatabase = FirebaseDatabase.getInstance()
        database.setPersistenceEnabled(true)
        return database
    }

    @Provides
    @Singleton
    fun provideDatabase(application: Application): Database {
       return Room.databaseBuilder(application, Database::class.java, "market-storage").build()
    }
}